### PR TITLE
cli (new): add `run packagemanifests`

### PR DESCRIPTION
**Pull request by estroz**
**Description of the change:** 
* cmd/operator-sdk: add `run packagemanifests` to new CLI, and set defaults for new project layouts
* test/e2e-new: test `run packagemanifests`
* test/utils: add utils to install/uninstall OLM

**Motivation for the change:** `run packagemanifests` should be available for new operator project layouts. This command must be used on operator projects with a package manifests format available.

/kind feature

### CODE REVIEW

**Comment by estroz** [comment]
This command will incidentally read `config/packagemanifests/bases` and assume it is a valid package, causing this command to error. #3257 fixes this as it moves the relevant package manifests files to a directory `<project root>/packagemanifests/` separate from kustomize-related files.

File: hack/tests/e2e-go-new.sh [hunk 1]
```diff
@@ -40,4 +40,4 @@ docker pull gcr.io/kubebuilder/kube-rbac-proxy:v0.5.0
 kind load docker-image gcr.io/kubebuilder/kube-rbac-proxy:v0.5.0
 
 trap_add cleanup EXIT
-go test -v $tests
+go test -timeout 20m -v $tests
```
Comment (joelanford) [inline hack/tests/e2e-go-new.sh:43]: Nit/question: is this change related to this PR? If not, could we make this in a separate PR?

File: internal/olm/operator/packagemanifests.go [hunk 2]
```diff
@@ -61,9 +62,14 @@ func (c *PackageManifestsCmd) validate() error {
 	if c.ManifestsDir == "" {
 		return errors.New("manifests dir must be set")
 	}
+	if _, err := os.Stat(c.ManifestsDir); err != nil {
+		return err
+	}
```
Comment (joelanford) [inline internal/olm/operator/packagemanifests.go:67]: Should we also check here that it's a directory?

File: test/utils/utils.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
+// Copyright 2020 The Operator-SDK Authors
```
Comment (joelanford) [inline test/utils/utils.go:1]: To limit our public API exposure leading up to a 1.0 release, would it make sense to put this in an internal directory for now so that it doesn't become part of our 1.0 API?

We _could_ always add it into the SDK library as a non-breaking new feature after 1.0 if users would find it helpful.

File: cmd/operator-sdk/run/packagemanifests/packagemanifests.go [hunk 4]
```diff
@@ -34,17 +35,23 @@ func NewCmd() *cobra.Command {
 	c := &packagemanifestsCmd{}
 
 	cmd := &cobra.Command{
-		Use:   "packagemanifests",
-		Short: "Run an Operator organized in the package manifests format with OLM",
+		Use: "packagemanifests",
+		Short: `Run an Operator in the package manifests format with OLM. This command will not work for Operators
+in a bundle format.`,
```
[the remark this answers is not in the document]
Comment (estroz) [reply cmd/operator-sdk/run/packagemanifests/packagemanifests.go:40]: No they can co-exist. The wording isn't great, how about
```suggestion
		Short: `Run an Operator in the package manifests format with OLM. This command will not work on
a bundle format.`,
```

File: hack/tests/e2e-go-new.sh (hunk 1 above)
Comment (estroz) [reply hack/tests/e2e-go-new.sh:43]: It is related. OLM takes a while to install :slightly_frowning_face:.

File: test/utils/utils.go (hunk 3 above)
Comment (estroz) [reply test/utils/utils.go:1]: I'd like to keep this internal, so I'll move it to `test/internal/utils`.

Comment (joelanford) [reply test/utils/utils.go:1]: probably needs to be `internal/test` so that it can be imported from anywhere in the SDK CLI repo.

File: cmd/operator-sdk/run/packagemanifests/packagemanifests.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (estroz) [reply cmd/operator-sdk/run/packagemanifests/packagemanifests.go:40]: Actually:
```
'run packagemanifests' deploys an Operator's package manifests with OLM. The command's argument
must be set to a valid package manifests root directory, ex. '<project-root>/packagemanifests'.
```

File: test/utils/utils.go (hunk 3 above)
Comment (joelanford) [reply test/utils/utils.go:1]: ^ Disregard. `test/internal` should be okay for `utils` since we only import it from within `test`.

File: cmd/operator-sdk/run/packagemanifests/packagemanifests.go [hunk 5]
```diff
@@ -35,16 +36,23 @@ func NewCmd() *cobra.Command {
 
 	cmd := &cobra.Command{
 		Use:   "packagemanifests",
-		Short: "Run an Operator organized in the package manifests format with OLM",
+		Short: "Deploy an Operator in the package manifests format with OLM",
+		Long: `'run packagemanifests' deploys an Operator's package manifests with OLM. The command's argument
```
Comment (tengqm) [inline cmd/operator-sdk/run/packagemanifests/packagemanifests.go:40]: if we have a choice between `run packagemanifests` and `run olm`, I'd vote for the latter one.

Comment (joelanford) [reply cmd/operator-sdk/run/packagemanifests/packagemanifests.go:40]: @tengqm the reason we're thinking of using `packagemanifests` is to distinguish it from an upcoming [`run bundle` subcommand](https://github.com/operator-framework/enhancements/blob/master/enhancements/run-bundle.md). The idea is that we want to support various OLM packaging formats for the `run` subcommand, and if we tried to encapsulate all of that under a single `run olm` command, the UX would get unwieldy since there are some differences in how they work.

Comment (tengqm) [reply cmd/operator-sdk/run/packagemanifests/packagemanifests.go:40]: Thanks for the info, @joelanford. I am just not a big fan of long commands, :).

Comment (joelanford) [reply cmd/operator-sdk/run/packagemanifests/packagemanifests.go:40]: @estroz Perhaps we could add a cobra alias for this (e.g. `operator-sdk run pm`)?

Comment (estroz) [reply cmd/operator-sdk/run/packagemanifests/packagemanifests.go:40]: SGTM, I'll make a follow-up.

Comment (estroz) [reply cmd/operator-sdk/run/packagemanifests/packagemanifests.go:40]: https://github.com/operator-framework/operator-sdk/pull/3314

File: cmd/operator-sdk/cli/legacy.go [hunk 6]
```diff
@@ -82,7 +82,7 @@ func GetCLIRoot() *cobra.Command {
 		new.NewCmd(),
 		olm.NewCmd(),
 		printdeps.NewCmd(),
-		run.NewCmd(),
+		run.NewCmdLegacy(),
```
Comment (camilamacedo86) [inline cmd/operator-sdk/cli/legacy.go:85]: helm/Ansible still requiring run `operator-sdk run --local` with the new layout.
